iamf/cli/codec/opus_encoder.cc

```diff
@@ -153,18 +153,21 @@ absl::Status OpusEncoder::InitializeEncoder() {
   // `OPUS_SET_BITRATE` treats this as the bit-rate for the entire substream.
   // Configure `libopus` so coupled substreams and mono substreams have equally
   // effective bit-rate per channel.
+  float opus_rate;
   if (num_channels_ > 1) {
-    opus_encoder_ctl(
-        encoder_,
-        OPUS_SET_BITRATE(static_cast<opus_int32>(
-            encoder_metadata_.target_bitrate_per_channel() * num_channels_ *
-                encoder_metadata_.coupling_rate_adjustment() +
-            0.5f)));
+    opus_rate =
+        encoder_metadata_.use_substream_rate_override()
+            ? encoder_metadata_.substream_rate_overrides()[substream_id_]
```

Comment:
This introduces an error if `encoder_metadata_.substream_rate_overrides()[substream_id_]` goes out of bounds.

Before performing the lookup validate it does not go out of bounds. At review time check against the size of the array. But if you take the "map"-based suggestion in another comment check that the substream_id is present.

I think if it would go out of bounds it is OK to fallback to the calculation based on `target_bitrate_per_channels`.

Same in the else.

Reply:
Thanks for the help, was considering it but was not sure how maps worked here.